Sources/Basic/SignalHandler.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import libc
+
+/// The single shared signal manager object, this is initialized here and not in class because we want to reference it from a C function pointer.
+private let sharedSignalManager = SignalManager()
+
+/// The old sigaction to chain the handlers of signals.
+private var oldAction = sigaction()
+
+/// A class to manage POSIX signals.
+public final class SignalManager {
```

Comment:
I'm not sure this makes sense... it isn't safe to do most things in signal handlers, so, e.g., the clearing of the callbacksMap inside the handler isn't safe.

I'm actually no longer sure that I know how to do this correctly... I was going to suggest having a single facility ensuring that we would get a signal at a particular time in the future, but I'm not sure how to guarantee that this signal causes the thread we want to be interrupted to be done so.

I think I'd need to do more research on how to manage this with POSIX before I'm comfortable implementing this... let me discuss with @abertelrud and see what he thinks we should do here.